rust/connlib/tunnel/src/device_channel/tun_android.rs

```diff
@@ -100,16 +100,15 @@ impl ioctl::Request<GetInterfaceNamePayload> {
     }
 
     fn name(&self) -> std::borrow::Cow<'_, str> {
-        // Safety: The memory of `self.name` is always initialized.
-        let cstr = unsafe { std::ffi::CStr::from_ptr(self.name.as_ptr() as _) };
-
-        cstr.to_string_lossy()
+        String::from_utf8_lossy(&self.name)
     }
 }
 
 #[derive(Default)]
 #[repr(C)]
-struct GetInterfaceNamePayload;
+struct GetInterfaceNamePayload {
+    name: [u8; libc::IF_NAMESIZE],
```

Comment:
I think it is very unlikely that this fixes the bug because you are not using this field. You'd have to access it via `self.payload.name`.

Reply:
If this actually fixes it, can you log how long the interface name is once it is retrieved?

Reply:
It's only 4 characters, flops between `tun0` and `tun1` as we rebuild the VpnService.

It's true that when this happens, the interface name is not null terminated -- but it's not anywhere in the name field. Reading that field causes the segfault.

IIRC The TUNGETIFF ioctl writes the interface name into the request parameter, not the payload (which is blank), unlike the mtu ioctl.